tests/System.Text.Primitives.Tests/Formatting/TimeTests.cs

```diff
@@ -0,0 +1,184 @@
+        {
+            TextFormat format = (formatChar == 0) ? default(TextFormat) : formatChar;
+            string expected = (format.IsDefault)
+                ? dto.ToString(CultureInfo.InvariantCulture)
+                : dto.ToString(formatChar.ToString(), CultureInfo.InvariantCulture);
+
+            var span = new Span<byte>(new byte[256]);
+            Assert.True(PrimitiveFormatter.TryFormat(dto, span, out int written, format, encoder));
+
+            var actual = TestHelper.SpanToString(span.Slice(0, written), encoder);
+            Assert.Equal(expected, actual);
+        }
+
+        [Theory]
+        [InlineData('c')]
+        [InlineData('t')]
+        [InlineData('g')]
+        [InlineData('G')]
+        public void SpecificTimeSpanTests(char format)
+        {
+            foreach (var encoder in Encoders)
+            {
+                TestTimeSpanFormat(TimeSpan.MinValue, format, encoder);
+                TestTimeSpanFormat(TimeSpan.MaxValue, format, encoder);
+                TestTimeSpanFormat(new TimeSpan(0), format, encoder);
+                TestTimeSpanFormat(new TimeSpan(-1), format, encoder);
+                TestTimeSpanFormat(new TimeSpan(1), format, encoder);
+                TestTimeSpanFormat(TimeSpan.FromDays(-1), format, encoder);
+                TestTimeSpanFormat(TimeSpan.FromDays(1), format, encoder);
+                TestTimeSpanFormat(TimeSpan.FromHours(-1), format, encoder);
+                TestTimeSpanFormat(TimeSpan.FromHours(1), format, encoder);
+                TestTimeSpanFormat(TimeSpan.FromMinutes(-1), format, encoder);
+                TestTimeSpanFormat(TimeSpan.FromMinutes(1), format, encoder);
+                TestTimeSpanFormat(TimeSpan.FromSeconds(-1), format, encoder);
+                TestTimeSpanFormat(TimeSpan.FromSeconds(1), format, encoder);
+            }
+        }
+
+        [Theory]
+        [InlineData('c')]
+        [InlineData('t')]
+        [InlineData('g')]
+        [InlineData('G')]
+        public void RandomTimeSpanTests(char format)
+        {
+            for (var i = 0; i < NumberOfRandomSamples; i++)
+            {
+                var ts = CreateRandomTimeSpan();
+                foreach (var encoder in Encoders)
+                {
+                    TestTimeSpanFormat(ts, format, encoder);
+                }
+            }
+        }
+
+        static void TestTimeSpanFormat(TimeSpan ts, char format, TextEncoder encoder)
+        {
+            var expected = ts.ToString(format.ToString(), CultureInfo.InvariantCulture);
+
+            var span = new Span<byte>(new byte[256]);
+            Assert.True(PrimitiveFormatter.TryFormat(ts, span, out int written, format, encoder));
+
+            var actual = TestHelper.SpanToString(span.Slice(0, written), encoder);
+            Assert.Equal(expected, actual);
+        }
+
+        static readonly Random Rnd = new Random();
```

Comment:
I think it makes sense to keep test runs consistent. If the test does catch a bug, we want it to be deterministic and always fail rather than seeing intermittent failures which could cause confusion (scenario: seeing a red in CI, re-test assuming some CI infra issue, and it goes green). If we want higher coverage, should we be explicit in our test cases and brute force a large input set?

> better coverage over time

I suppose that would be OK but imo being explicit in the edge cases/etc would be better.